server/src/datasources/SQLDataSource.ts

```diff
@@ -0,0 +1,96 @@
+import {DataSource, DataSourceConfig} from 'apollo-datasource';
+import Knex, {QueryBuilder} from "knex";
+import snakeCase from "lodash.snakecase";
+import snakecaseKeys from "snakecase-keys";
+import SelectionTree from "../types/SelectionTree";
+
+export interface SQLDataSourceConfig {
```

Comment:
Style nitpick: why all caps `SQLDataSourceConfig` instead of `SqlDataSourceConfig`? 

Also, is this abstraction on top of `knex` necessary?

Reply:
[I follow the name style as: `RESTDataSource`](https://www.apollographql.com/docs/apollo-server/data/data-sources/#rest-data-source). It would be fine if we choose different name style for abbreviations and initials.

This abstraction is necessary or else there would be a lot duplicated code between datasource for different resources.